### ISSUE TITLE
Date codec

### ISSUE THREAD

**Issue by Kinrany** [gcanti/io-ts#504]
## 🚀 Feature request

A codec that represents a `Date` without de-/serialization.

### Current Behavior

`io-ts-types` has several codecs that deserialize from strings and numbers, but there is no codec that represents a date.

### Suggested Solution

Add a codec that represents this standard type to `io-ts` itself.

### Who does this impact? Who is this for?

All users

### Describe alternatives you've considered

It could be added to `io-ts-types` instead or written inline, but it seems like a basic enough feature to be in the core library.

### Your environment

| Software   | Version(s) |
| ---------- | ---------- |
| io-ts      |       2.2.3     |
| fp-ts      |      2.6.1      |
| TypeScript |    3.9.3        |

**Comment by mlegenhausen** [comment]
It does already exist: https://github.com/gcanti/io-ts-types/blob/master/src/date.ts

**Comment by Kinrany** [comment]
It does, but it's not public, is it?

**Comment by gcanti** [comment]
@Kinrany 

```ts
import { date } from 'io-ts-types/lib/date'
```

or (v0.5.8+)

```ts
import { date } from 'io-ts-types'
```

**Comment by Kinrany** [comment]
Ooh, right. I expected it to be in CamelCase for some reason. Thanks! 16:11, July 29, 2020, Giulio Canti <notifications@github.com>:
Closed #504.
